ogr/ogrsf_frmts/generic/ogrlayerpool.cpp

```diff
@@ -444,19 +448,21 @@ OGRFeatureDefn *OGRProxiedLayer::GetLayerDefn()
 /*                            GetSpatialRef()                           */
 /************************************************************************/
 
-OGRSpatialReference *OGRProxiedLayer::GetSpatialRef()
+const OGRSpatialReference *OGRProxiedLayer::GetSpatialRef() const
 {
+    std::lock_guard oLock(m_oMutex);
     if (poSRS != nullptr)
         return poSRS;
     if (poUnderlyingLayer == nullptr && !OpenUnderlyingLayer())
         return nullptr;
-    OGRSpatialReference *poRet = poUnderlyingLayer->GetSpatialRef();
-    if (poRet != nullptr)
+    OGRSpatialReference *l_poSRS =
+        const_cast<OGRSpatialReference *>(poUnderlyingLayer->GetSpatialRef());
+    if (l_poSRS != nullptr)
     {
-        poSRS = poRet;
-        poSRS->Reference();
+        l_poSRS->Reference();
```

Comment:
I wonder if making the reference-counting functions `const` is better than const-casting the return values here.

Reply:
I was considering that too, but I'm a bit hesitatant, because there is a GetRefCount() method. So Reference()/Dereference() have observable side-effects, and thus are normally not candidate for being const